server/cherry_pick.go

```diff
@@ -316,15 +328,65 @@ func (s *Server) addAssignee(ctx context.Context, newPRNumber int, pr *model.Pul
 	}
 }
 
-func returnToMaster(dir string) {
-	cmd := exec.Command("git", "checkout", "master")
+func returnToMaster(ctx context.Context, dir string) error {
+	cmd := exec.CommandContext(ctx, "git", "checkout", "master")
+	if err := runCommand(cmd, dir); err != nil {
+		return err
+	}
+	return nil
+}
+
+func cloneRepo(ctx context.Context, dir, upstreamSlug, originSlug, repoName string) error {
+	// Set username and email first.
+	cmd := exec.CommandContext(ctx, "git", "config", "user.name")
+	if out, err := runCommandWithOutput(cmd, dir); err != nil {
+		return err
+	} else if out == "" { // this means username is not set
+		cmd = exec.CommandContext(ctx, "git", "config", "--global", "user.name", "mattermost-build")
```

Comment:
These need to be checked and set per cloned repository, `--global` gitconfig is not accessible.
And because of this I think you need to move this block (`git config user.{name|email}`) after the repos is cloned and inside the cloned local folder.

Reply:
Ah my bad. Will change.

Reply:
`cd <just-cloned-folder>` command seems to be missing now.

Reply:
It's not. `cmd.Dir = dir` is set and dir is `filepath.Join(dir, repoName)` ;)